File: SelectSort.cpp
#include<iostream>
using namespace std;
void SelectSort(int *a,int n){
    for(int i=1;i<=n;i++){
        int pos=i;
        for(int j=i;j<=n;j++){
            if(a[j]<a[pos]){
                pos=j;
            }
        }
        if(a[pos]<a[i])
            swap(a[pos],a[i]);
    }
}
int main(){
    int n;
    cin>>n;
    int a[n+1];
    for(int i=1;i<=n;i++)
        cin>>a[i];
    SelectSort(a,n);
    for(int i=1;i<=n;i++)
        cout<<a[i]<<' ';
    return 0;
}

File: SelectSort.go
package main
import (
	"fmt"
)
func SelectSort(a []int,n int){
	for i:=0;i<n;i++{
		pos := i
		for j:=i;j<n;j++{
			if a[pos]>a[j] {
				pos=j
			}
		}
		if a[pos]<a[i]{
			a[i] , a[pos] = a[pos] , a[i]
		}
	}
}
func main(){
	var a []int
	var n,x int
	fmt.Scanln(&n)
	for i:=0;i<n;i++{
		fmt.Scan(&x)
		a = append(a,x)
	}
	SelectSort(a,n)
	fmt.Println(a)
}